runners/direct-java/src/main/java/org/apache/beam/runners/direct/DirectRunner.java

```diff
@@ -392,4 +411,82 @@ public State waitUntilFinish(Duration duration) throws IOException {
           "DirectPipelineResult does not support waitUntilFinish.");
     }
   }
+
+  /**
+   * Internal configuration options for the {@link DirectRunner}. Deals with internal implementation
+   * details.
+   */
+  public interface InternalOptions extends DirectOptions {
+    /**
+     * Gets the {@link ExecutorServiceFactory} to use to create instances of {@link ExecutorService}
+     * to execute {@link PTransform PTransforms}.
+     *
+     * <p>Defaults to a {@link FixedThreadPoolSupplier}, which produces instances of
+     * {@link Executors#newFixedThreadPool(int)}.
+     */
+    @JsonIgnore
+    @Required
+    @Hidden
+    @Default.InstanceFactory(FixedThreadPoolSupplier.class)
+    OptionsSupplier<ExecutorService> getExecutorServiceFactory();
+
+    void setExecutorServiceFactory(OptionsSupplier<ExecutorService> executorService);
```

Comment:
As a future thought, would it be good to consolidate executors, I believe there is more than one options class now that does this.